packages/configs/default/index.json

```diff
@@ -4,12 +4,13 @@
     "types:*.{ts,tsx}": ["@parcel/transformer-typescript-types"],
     "bundle-text:*": ["@parcel/transformer-inline-string", "..."],
     "data-url:*": ["@parcel/transformer-inline-string", "..."],
-    "*.{js,mjs,jsm,jsx,es6,cjs,ts,tsx}": [
+    "*.{js,mjs,jsm,jsx,es6,cjs,ts-babel,tsx-babel}": [
```

Comment:
Not a fan of this `ts-babel` extension, seems like a hack. These transformers should not depend on each other, it kind of defeats the purpose of having the plugins as seperate entities.

Reply:
Note that this would also just end up with using tsc by default on 99% of Typescript projects as every typescript project should have a tsconfig anyway... otherwise why are you using Typescript in the first place if you don't do typechecking...

Reply:
Yeah I honestly don't want to do this either. But there's literally no other way to make babel process it instead of TSC in such situations.

Typechecking must be integrated into the bundler for a variety of reasons, and one of them is that Vue TypeScript can't be processed.

I'm considering just dropping this PR altogether but I really, really want TypeScript to be checked by default before the build finishes.

Reply:
I know that most TypeScript projects would end up using `tsc` because of this but that's kind of the point; few TypeScript users actually want to use Babel, but Parcel can still be called "blazing-fast" if you decide to use Babel.

Reply:
Like @mischnic already mentioned adding `@parcel/transformer-typescript-tsc` won't do anything more than using Babel, typechecking is done by the validator plugin... and it also doesn't support vue (yet)

Reply:
Also Devon mentioned in the Typescript issue even if we wanted to enable type-checking by default the performance and reliability of our typechecking validator is not good enough to do so, mainly because tsc isn't really built for what we use it but we can work around most of that.

Fixing most issues with the typechecker diagnostics plugin seems more important than this, which is just a config change and removal of tests.

So if you really want this to be the default, help us ensure our typescript transformer and validator are top-notch so we don't have to worry about the various bugs we encountered in Parcel 1 and performance issues we might experience now.

The transformer mainly had too much freedom in Parcel 1 and probably still does in Parcel 2, we should ignore probably 90% of compiler options and only whitelist a few we want to allow, so it does not break any further compilation down the line.

Besides this we also want to support the paths field in jsconfig and tsconfig in our resolver at some point and want typechecking to be faster. We've even thought about a command like `parcel validate` to just run the validators without any compilation to run linting and typechecking a lot faster than the tools can do themselves by only running over changed files

Reply:
I'll make more useful contributions instead of this PR given your feedback. Closing.